File: bypass.py
"""
ShrinkMe.click Bypass Script
Resolved destination URL using cloudscraper + form submission

Usage:
    python bypass.py https://shrinkme.click/pRgC7uL
    python bypass.py (uses default URL)
"""

import cloudscraper
from bs4 import BeautifulSoup
import time
import sys
import json

def bypass_shrinkme(url):
    print(f"\n[.Target] {url}")
    
    url = url.rstrip('/')
    code = url.split("/")[-1]
    print(f"[.Code] {code}")
    
    client = cloudscraper.create_scraper(allow_brotli=False)
    
    domains = [
        "https://en.shrinke.me",
        "https://shrinkme.click",
        "https://shrinke.me",
        "https://shrinkme.io"
    ]
    
    for domain in domains:
        try:
            target_url = f"{domain}/{code}"
            print(f"\n[.Trying] {domain}")
            
            headers = {
                "referer": "https://mrproblogger.com/",
                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
            }
            
            resp = client.get(target_url, headers=headers)
            
            if resp.status_code != 200:
                continue
            
            soup = BeautifulSoup(resp.content, "html.parser")
            inputs = soup.find_all("input")
            data = {inp.get('name'): inp.get('value', '') for inp in inputs if inp.get('name')}
            
            if not data:
                continue
            
            print(f"[.Wait] 15s...")
            time.sleep(15)
            
            go_headers = {
                "x-requested-with": "XMLHttpRequest",
                "referer": target_url,
                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
            }
            
            r = client.post(f"{domain}/links/go", data=data, headers=go_headers)
            result = r.json()
            
            if 'url' in result:
                final_url = result['url']
                print(f"\n{'='*60}")
                print(f"[+] DESTINATION URL:")
                print(f"    {final_url}")
                print(f"{'='*60}")
                return final_url
                
        except Exception as e:
            continue
    
    print("\n[-] All domains failed")
    return None

if __name__ == "__main__":
    url = sys.argv[1] if len(sys.argv) > 1 else "https://shrinkme.click/pRgC7uL"
    result = bypass_shrinkme(url)
    if result:
        print(f"\n[+] Copy: {result}")

File: api/bypass.js
const axios = require('axios');
const cheerio = require('cheerio');

module.exports = async (req, res) => {
  // CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  const { url } = req.query;

  if (!url) {
    return res.status(400).json({ 
      error: 'Missing url parameter',
      usage: '?url=https://shrinkme.click/xxxxx'
    });
  }

  try {
    const result = await bypassShrinkme(url);
    return res.status(200).json(result);
  } catch (error) {
    return res.status(500).json({ error: error.message });
  }
};

async function bypassShrinkme(url) {
  const code = url.replace(/\/$/, '').split('/').pop();
  
  const domains = [
    'https://en.shrinke.me',
    'https://shrinkme.click',
    'https://shrinke.me'
  ];

  const client = axios.create({
    timeout: 30000,
    headers: {
      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
      'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
      'Accept-Language': 'en-US,en;q=0.5',
    }
  });

  for (const domain of domains) {
    try {
      const targetUrl = `${domain}/${code}`;
      
      // GET request with referer
      const resp = await client.get(targetUrl, {
        headers: { 'referer': 'https://mrproblogger.com/' }
      });

      if (resp.status !== 200) continue;

      // Parse form data
      const $ = cheerio.load(resp.data);
      const formData = {};
      $('input').each((_, el) => {
        const name = $(el).attr('name');
        const value = $(el).attr('value') || '';
        if (name) formData[name] = value;
      });

      if (Object.keys(formData).length === 0) continue;

      // Wait 12 seconds (Vercel timeout is 10s for hobby, 60s for pro)
      await new Promise(resolve => setTimeout(resolve, 12000));

      // POST to /links/go
      const goResp = await client.post(`${domain}/links/go`, 
        new URLSearchParams(formData).toString(),
        {
          headers: {
            'x-requested-with': 'XMLHttpRequest',
            'referer': targetUrl,
            'content-type': 'application/x-www-form-urlencoded'
          }
        }
      );

      const result = goResp.data;
      
      if (result && result.url) {
        return {
          success: true,
          source: url,
          destination: result.url,
          message: result.message || 'OK'
        };
      }

    } catch (e) {
      continue;
    }
  }

  throw new Error('All bypass methods failed');
}
